pandas/tests/sparse/test_frame.py

```diff
@@ -1118,6 +1118,31 @@ def test_isnotnull(self):
                             'B': [True, False, True, True, False]})
         tm.assert_frame_equal(res.to_dense(), exp)
 
+    def test_from_to_scipy(self):
+        # GH 4343
+        try:
+            from scipy.sparse import csr_matrix
+        except ImportError:
+            return  # scipy not available
+
+        arr = np.array([[0, 1, 0],
+                        [0, 0, 1],
+                        [1, 1, 1.]])
+        spm = csr_matrix(arr)
+
+        for index, columns in ((list('abc'), list('def')),
+                               (None, None)):
+            sdf = pd.SparseDataFrame(spm, index=index, columns=columns)
+
```

Comment:
I would rather you actually construct the SDF directly and simply use ``tm.assert_sp_frame_equal`` which already does all of this.

Further check for various fill types as well.

I would write this using ``pytest.parametrize`` (as our new way of doing things)